### PR TITLE
Updating test cases to consider issue #274 - Problem with required an…

### PR DESCRIPTION
**Pull request by aguirrel**

### CODE REVIEW

File: cerberus/tests/test_validation.py [hunk 1]
```diff
@@ -840,6 +840,7 @@ def test_dependencies_errors():
                 error=('field2', ('field2', 'dependencies'),
                        errors.DEPENDENCIES_FIELD_VALUE,
                        {'field1': ['one', 'two']}, ({'field1': 'three'},)))
+    assert_success({'field1': 'three'}, validator=v)
```
Comment (funkyfuture) [inline cerberus/tests/test_validation.py]: this should not succeed as `field2` is defined to be required.

Comment (aguirrel) [inline cerberus/tests/test_validation.py]: I have just modified source code to be compliant with docs.  All test cases are working now, included this one ;)

**Comment by funkyfuture** [comment]
my impression was rather that the docs should be clarified. but that should be discussed in #229.

i'd had some minor remarks here. and, obviously, the docs would have to be amended if this was changed in a very elaborative and exemplary way. :-)

**Comment by aguirrel** [comment]
ok I understand. Please, correct me if I'm wrong, but I think people base theirs development on docs (at least two persons did ;) ).  Indeed I think is a very consistent way to manage 'required' validator.  In fact this new test case worked on version 0.9 but not in 0.9.1.

I think merging this PR is like "fixing" an issue but if you rewrite docs it could be seem a little odd.

Personally I suggest not changing so much in docs in version 1.0.x and change it in 1.1.x.

**Comment by funkyfuture** [comment]
afaik, the behavior you expect was never intended, and as pointed out in the correspoinding issue, the docs can be interpreted like you do, but are not meant to. for an informed discussion it would be good to know which commit changed that behaviour. `git bisect` is great for that.

**Comment by aguirrel** [comment]
Ok I understand.  Please feel free to to close this PR without merging.  Anyway I can't do it work in all situations.  If in the test case, you just invert the order of the tests (first success and then fail) it just doesn't work.  It seems that this PR only works if the field is present in the document you are validating or you name it in other validator call (as is happening in the test case).

**Pull request closed without merging**